pinot-common/src/main/java/com/linkedin/pinot/common/segment/SegmentMetadata.java

```diff
@@ -15,158 +15,98 @@
-   */
-  public String getName();
+  String getName();
 
-  /**
-   * @return
-   */
-  public long getIndexCreationTime();
+  long getIndexCreationTime();
 
   /**
-   * Returns the last time that this segment was pushed or Long.MIN_VALUE if it has never been
-   * pushed.
+   * Get the last time that this segment was pushed or <code>Long.MIN_VALUE</code> if it has never been pushed.
    */
-  public long getPushTime();
+  long getPushTime();
 
   /**
-   * Returns the last time that this segment was refreshed or Long.MIN_VALUE if it has never been
-   * refreshed.
+   * Get the last time that this segment was refreshed or <code>Long.MIN_VALUE</code> if it has never been refreshed.
    */
-  public long getRefreshTime();
+  long getRefreshTime();
 
-  /**
-   * Returns if a column has dictionary or not.
-   */
-  public boolean hasDictionary(String columnName);
+  boolean hasDictionary(String columnName);
 
-  /** Returns true if the segment has a StarTree index defined */
-  public boolean hasStarTree();
+  boolean hasStarTree();
 
-  /**
-   * Returns the StarTreeMetadata for the segment
-   * @return
-   */
-  public StarTreeMetadata getStarTreeMetadata();
+  @Nullable
+  StarTreeMetadata getStarTreeMetadata();
 
   /**
-   * returns the forward Index file name with appropriate extension for a given version
-   * @param column
-   * @return
+   * Get the forward index file name with appropriate extension for a given version.
+   *
+   * @param column column name.
+   * @param segmentVersion segment version.
+   * @return forward index file name.
    */
   String getForwardIndexFileName(String column, String segmentVersion);
 
   /**
-   * returns the dictionary file name with appropriate extension for a given version
-   * @param column
-   * @return
+   * Get the dictionary file name with appropriate extension for a given version.
+   *
+   * @param column column name.
+   * @param segmentVersion segment version.
+   * @return dictionary file name.
    */
   String getDictionaryFileName(String column, String segmentVersion);
 
   /**
-   * returns the bitmap inverted index file name with appropriate extension for a given version
-   * @param column
-   * @return
+   * Get the bitmap inverted index file name with appropriate extension for a given version.
+   *
+   * @param column column name.
+   * @param segmentVersion segment version.
+   * @return bitmap inverted index file name.
    */
   String getBitmapInvertedIndexFileName(String column, String segmentVersion);
 
-  /**
-   * returns the name of the component that created the segment
-   * @return
-   */
   @Nullable
   String getCreatorName();
 
-  /**
-   * returns the padding character
-   * @return
-   */
-  Character getPaddingCharacter();
+  char getPaddingCharacter();
 
-  /**
-   * returns Hll Log2m parameter
-   * @return
-   */
   int getHllLog2m();
 
-  /**
-   * @return
-   */
-  public Map<String, String> toMap();
+  Map<String, String> getHllDerivedColumnMap();
```

Comment:
Use getHllDerivedColumnFor(String) -> String instead of letting the full map out.
Reason is, we are planning more derived columns going forward  which will change internal organization. Right interface for that is
getDerivedColumn(columnName, derivedFieldType). You can do this or if, it's bigger change, then just provide getHllDerivedColumnFor(columnName), which is more manageable if class internals change.